src/tests/ftest/launch.py

```diff
@@ -52,2294 +68,2973 @@
-
-    if args.jenkinslog:
-        xml_file = os.path.join(new_test_logs_dir, "results.xml")
-        try:
-            with open(xml_file) as xml_buffer:
-                xml_data = xml_buffer.read()
-        except OSError as error:
-            log("Error reading {} : {}".format(xml_file, str(error)))
-            status |= 1024
-            return status
+    def execute(self, test, repeat, sparse, fail_fast, extra_yaml):
+        """Run the specified test.
 
-        # save it for the Launchable [de-]mangle
-        org_xml_data = xml_data
+        Args:
+            test (TestInfo): the test information
+            repeat (int): the test repetition number
+            sparse (bool): whether to use avocado sparse output
+            fail_fast(bool): whether to use the avocado fail fast option
+            extra_yaml (list): whether to use an exta yaml file with the avocado run command
 
-        # First, mangle the in-place file for Jenkins to consume
-        test_dir = os.path.split(os.path.dirname(test_file))[-1]
-        org_class = "classname=\""
-        new_class = "{}FTEST_{}.".format(org_class, test_dir)
-        xml_data = re.sub(org_class, new_class, xml_data)
+        Returns:
+            int: status code: 0 = success, >0 = failure
 
-        try:
-            with open(xml_file, "w") as xml_buffer:
-                xml_buffer.write(xml_data)
-        except OSError as error:
-            log("Error writing {}: {}".format(xml_file, str(error)))
-            status |= 1024
-
-        # Now mangle (or rather unmangle back to canonical xunit1 format)
-        # another copy for Launchable
-        xml_data = org_xml_data
-        org_name = r'(name=")\d+-\.\/.+.(test_[^;]+);[^"]+(")'
-        new_name = r'\1\2\3 file="{}"'.format(test_file)
-        xml_data = re.sub(org_name, new_name, xml_data)
-        xml_file = xml_file[0:-11] + "xunit1_results.xml"
+        """
+        logger.debug("=" * 80)
+        command = self.avocado.get_run_command(
+            test, self.tag_filters, sparse, fail_fast, extra_yaml)
+        logger.info(
+            "Running the %s test on repeat %s/%s: %s", test, repeat, self.repeat, " ".join(command))
+        start_time = int(time.time())
 
         try:
-            with open(xml_file, "w") as xml_buffer:
-                xml_buffer.write(xml_data)
-        except OSError as error:
-            log("Error writing {}: {}".format(xml_file, str(error)))
-            status |= 1024
-    return status
+            return_code = run_local(logger, command, capture_output=False, check=False).returncode
+            if return_code == 0:
+                logger.debug("All avocado test variants passed")
+            elif return_code == 2:
+                logger.debug("At least one avocado test variant failed")
+            elif return_code == 4:
+                message = "Failed avocado commands detected"
+                self._fail_test(self.result.tests[-1], "Process", message)
```

Comment:
(pylint-no-value-for-parameter) No value for argument 'fail_reason' in staticmethod call